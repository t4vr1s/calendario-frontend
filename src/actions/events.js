import { fetchConToken } from '../helpers/fetch'
import { types } from '../types/types'
import { prepareEvents } from '../helpers/prepareEvents'
import Swal from 'sweetalert2'

export const eventStartAddNew = (event) => {
  return async (dispatch, getState) => {
    const { uid, name } = getState().auth
    try {
      const resp = await fetchConToken('events', event, 'POST')
      const body = await (await resp).json()
      console.log(body)
      console.log(event)
      if (body.ok) {
        event.id = body.evento.id
        event.user = {
          _id: uid,
          name: name
        }
        console.log(event)
        dispatch(eventAddNew(event))
      }
    } catch (error) {
      console.log(error)
    }
  }
}

export const eventAddNew = (event) => ({
  type: types.eventAddNew,
  payload: event
})

export const eventSetActive = (event) => ({
  type: types.eventSetActive,
  payload: event
})

export const eventClearActive = () => ({
  type: types.eventClearActive
})

export const eventStartUpdate = (event) => {
  return async (dispatch) => {
    try {
      const resp = await fetchConToken(`events/${event.id}`, event, 'PUT')
      const body = await resp.json()
      if (body.ok) {
        dispatch(eventUpdated(event))
      } else {
        Swal.fire('Error', body.msg, 'error')
      }
    } catch (error) {
      console.log(error)
    }
  }
}

const eventUpdated = (event) => ({
  type: types.eventUpdated,
  payload: event
})

export const eventStartDelete = () => {
  return async (dispatch, getState) => {
    const { id } = getState().calendar.activeEvent
    try {
      const resp = await fetchConToken(`events/${id}`, {}, 'DELETE')
      const body = await resp.json()
      if (body.ok) {
        dispatch(eventDeleted())
      } else {
        Swal.fire('Error', body.msg, 'error')
      }
    } catch (error) {
      console.log(error)
    }
  }
}

const eventDeleted = () => ({
  type: types.eventDeleted
})

export const eventStartLoading = () => {
  return async (dispatch) => {
    try {
      const resp = await fetchConToken('events')
      const body = await resp.json()
      const eventos = prepareEvents(body.eventos)
      dispatch(eventLoaded(eventos))
    } catch (error) {
      console.log(error)
    }
  }
}

const eventLoaded = (eventos) => ({
  type: types.eventLoaded,
  payload: eventos
})

export const eventLogout = () => ({
  type: types.eventLogout
})
