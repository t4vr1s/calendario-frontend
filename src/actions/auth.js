import Swal from 'sweetalert2'
import { fetchSinToken, fetchConToken } from '../helpers/fetch'
import { types } from '../types/types'
import { eventLogout } from './events'

export const starLogin = (email, password) => {
  return async (dispatch) => {
    const resp = await fetchSinToken('auth', { email, password }, 'POST')
    const body = await resp.json()
    const { ok, name, uid, token, msg } = body
    if (ok) {
      localStorage.setItem('token', token)
      localStorage.setItem('token-init-date', new Date().getTime())

      dispatch(
        login({
          uid: uid,
          name: name
        })
      )
    } else {
      Swal.fire('Error', msg, 'error')
    }
  }
}

export const startRegister = (name, email, password) => {
  return async (dispatch) => {
    const resp = await fetchSinToken(
      'auth/new',
      { name, email, password },
      'POST'
    )
    const body = await resp.json()
    const { ok, name: nameUser, uid, token, msg } = body
    if (ok) {
      localStorage.setItem('token', token)
      localStorage.setItem('token-init-date', new Date().getTime())

      dispatch(
        login({
          uid: uid,
          name: nameUser
        })
      )
    } else {
      Swal.fire('Error', msg, 'error')
    }
  }
}

export const startChecking = () => {
  return async (dispatch) => {
    const resp = await fetchConToken('auth/renew')
    const body = await resp.json()
    const { ok, name: nameUser, uid, token } = body
    if (ok) {
      localStorage.setItem('token', token)
      localStorage.setItem('token-init-date', new Date().getTime())

      dispatch(
        login({
          uid: uid,
          name: nameUser
        })
      )
    } else {
      dispatch(checkingFinish())
    }
  }
}

const checkingFinish = () => ({
  type: types.authCheckingFinish
})

const login = (user) => ({
  type: types.authLogin,
  payload: user
})

export const startLogout = () => {
  return (dispatch) => {
    localStorage.clear()
    dispatch(eventLogout())
    dispatch(logout())
  }
}

const logout = () => ({
  type: types.authLogout
})
