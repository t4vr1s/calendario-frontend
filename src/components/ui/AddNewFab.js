import React from 'react';
import { useDispatch } from 'react-redux';
import { uiOpenModal } from '../../actions/ui';

export const AddNewFab = () => {
  const dispatch = useDispatch();

  const handleClickModal = () => {
    dispatch(uiOpenModal());
  };

  return (
    <button onClick={handleClickModal} className="btn btn-primary fab">
      <i className="fas fa-plus"></i>
    </button>
  );
};
