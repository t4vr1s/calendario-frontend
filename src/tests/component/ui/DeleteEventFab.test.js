import React from 'react'
import { Provider } from 'react-redux'
import { mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import '@testing-library/jest-dom'
import { DeleteEventFab } from '../../../components/ui/DeleteEventFab'
import { eventStartDelete } from '../../../actions/events'

jest.mock('../../../actions/events', () => ({
  eventStartDelete: jest.fn()
}))

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initState = {}
const store = mockStore(initState)
store.dispatch = jest.fn()

const wrapper = mount(
  <Provider store={store}>
    <DeleteEventFab />
  </Provider>
)

describe('pruebas al componente <DeleteEventFab />', () => {
  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('debe llamar al onClick=eventStartDelete', () => {
    wrapper.find('button').prop('onClick')()
    expect(eventStartDelete).toHaveBeenCalled()
  })
})
