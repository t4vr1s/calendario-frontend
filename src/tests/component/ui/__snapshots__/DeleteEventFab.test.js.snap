// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas al componente <DeleteEventFab /> debe mostrarse correctamente 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
    Borrar evento
  </span>
</button>
`;
